src/textual/_animator.py

```diff
@@ -93,9 +95,19 @@ class SimpleAnimation(Animation):
     final_value: object
     easing: EasingFunction
     on_complete: CallbackType | None = None
+    animate_on_level: AnimationsEnum = AnimationsEnum.BASIC
+    """Minimum level required for the animation to take place (inclusive)."""
+
+    def __post_init__(self) -> None:
+        """Cache baseline that determines whether an animation runs."""
+        # We cache this value so we don't have to `.get` the context var repeatedly.
+        self.show_animations = active_app.get().show_animations
```

Comment:
I can add a fallback to fetch the env var in case there is no active app.
As for copying it from the app, we are caching it to prevent accessing the context var every single time we compute the next “frame” of the animation.

Reply:
My reply is no longer relevant after a first pass through the comments.